merkledag/merkledag.go

```diff
@@ -460,16 +443,24 @@ func EnumerateChildrenAsync(ctx context.Context, ds DAGService, c *cid.Cid, visi
 			if inProgress == 0 && next == nil {
 				return nil
 			}
-		case nc := <-out:
-			if nc.Err != nil {
-				return nc.Err
+		case l := <-out:
+			n, err := ds.Get(ctx, l)
+			if err != nil {
+				return err
 			}
 
 			if next == nil {
-				next = nc.Node
-				send = feed
+				if len(n.Links()) > 0 {
+					next = n.Links()[0].Cid
+					for _, l := range n.Links()[1:] {
```

Comment:
It might be beneficial to: preallocate array for Cids in `.Links()` call, write the Cids to that array and do just one append call. 

This way the todobuffer will be expanded just once instead of N times.